model_api/cpp/models/src/model_base.cpp

```diff
@@ -131,3 +131,11 @@ std::shared_ptr<ov::Model> ModelBase::getModel() {
     updateModelInfo();
     return model;
 }
+
+std::shared_ptr<InferenceAdapter> ModelBase::getInferenceAdapter() {
+    if (!inferenceAdapter) {
+        throw std::runtime_error(std::string("Model was created without inference adapter"));
```

Comment:
```suggestion
        throw std::runtime_error("Model wasn't loaded");
```
`inferenceAdapter` is always used. But when model path is used for construction, one needs to call `load()` to initialize adapter